crates/loader/src/local/mod.rs

```diff
@@ -225,6 +220,28 @@ async fn core(
     })
 }
 
+/// Ensures that the path is canonicalized and absolutized base on the `src` path
+fn canonicalize_and_absolutize(mut path: PathBuf, src: &Path) -> anyhow::Result<PathBuf> {
+    path = path.canonicalize().unwrap_or(path);
+    // If path is UTF-8 and we can successfully perform tilde and environment expansion, do so.
+    if let Some(p) = path.as_os_str().to_str() {
+        match shellexpand::full(p) {
+            Ok(p) => path = Path::new(&*p).to_owned(),
+            Err(e) => {
+                return Err(e.cause).context(anyhow!(
+                    "failed to expand `source` path: `${}` could not be expanded",
+                    e.var_name
+                ))
+            }
+        }
+    }
+    Ok(if path.is_absolute() {
+        path
+    } else {
+        src.join(path)
+    })
```

Comment:
I believe `Path::join` already does this: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=09ffc9d54f48c56c5476c67d549c6ca8

```suggestion
    Ok(src.join(path))
```